jetbrains-rider/tst/software/aws/toolkits/jetbrains/services/lambda/completion/DotNetHandlerCompletionTest.kt

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.completion
+
+import com.jetbrains.rd.framework.impl.RpcTimeouts
+import com.jetbrains.rider.model.ImageSourceIconModel
+import com.jetbrains.rider.model.lambdaModel
+import com.jetbrains.rider.projectView.solution
+import com.jetbrains.rider.test.annotations.TestEnvironment
+import com.jetbrains.rider.test.base.BaseTestWithSolution
+import org.assertj.core.api.Assertions.assertThat
+import org.testng.annotations.Test
+import software.amazon.awssdk.services.lambda.model.Runtime
+
+class DotNetHandlerCompletionTest : BaseTestWithSolution() {
+
+    override fun getSolutionDirectoryName(): String = ""
+
+    override val waitForCaches = true
+
+    @Test
+    @TestEnvironment(solution = "SamHelloWorldApp")
+    fun testCompletion_IsSupportedForDotNetRuntime() {
+        val provider = HandlerCompletionProvider(project, Runtime.DOTNETCORE2_1)
+        assertThat(provider.isCompletionSupported).isTrue()
+    }
+
+    @Test
+    @TestEnvironment(solution = "SamHelloWorldApp")
+    fun testDetermineHandlers_SingleHandler() {
+        val handlers = project.solution.lambdaModel.determineHandlers.sync(Unit, RpcTimeouts.default)
+
+        assertThat(handlers.size).isEqualTo(1)
+        assertThat(handlers.first().handler).isEqualTo("HelloWorld::HelloWorld.Function::FunctionHandler")
+        assertThat(handlers.first().iconId).isEqualTo(ImageSourceIconModel(iconPackStringId = 20, iconNameStringId = 31))
```

Comment:
Where do these icon numbers come from? How fragile are they?

Reply:
Also, are there images in Rider in the completion provider?

Reply:
This is a value from backend. We are getting a handler method psi element that has an image object for this element. It is different for c# method, F# method, static method, field, etc. It should have defined ids per element type.

Reply:
Is there a static variable that defines it as 20 we can reference? What are the chances it changes?

Reply:
We get only icon id from backend, then on a frontend we use this ID to get an icon from id-to-path map. We do not have any user readable static instances for ID that we get from backend. The ID should not change, it is a very low possibility it will happens once. I can create a test enum describing the icon id to make it easy to read.

> Also, are there images in Rider in the completion provider?
Sorry, not sure I understand the question. Those images are for each individual completion item. Rider plugin will have a completion support and populate completion results in DotNetHandlerCompletion.

Reply:
@abrooksv, updated tests to check for frontend icon path instead of raw ids.
> Also, are there images in Rider in the completion provider?

Sorry, not sure I got the question. In Rider we have similar completion for Run static method, but it does not have icons. We discussed internally that we should add similar logic to Rider to have icons here.

Reply:
I didn't remember seeing icons in the completion pop up so wasn't sure what they were for, that's all.

Reply:
Got it. Text editor completion popup has icons for completion items. We have similar run configuration with completion fields that specify a static method to run that does not have icons for now (but I have plans to add this logic in 2019.3).